scene/gui/tree.cpp

```diff
@@ -1396,6 +1425,7 @@ TreeItem::TreeItem(Tree *p_tree) {
 
 TreeItem::~TreeItem() {
 	_unlink_from_tree();
+	validate_cache();
```

Comment:
I did not.  But as mentioned in the description, this debug code is temporary while @KoBeWi is refactoring this code.  I put this check here in the destructor in case there were any operations I failed to instrument, so I would at least detect the inconsistency at the end (in the only function that definitely gets called in all versions of the code.)